be/src/exec/spill/dir_manager.cpp

```diff
@@ -14,25 +14,66 @@
 
 #include "exec/spill/dir_manager.h"
 
+#include <regex>
+
 #include "common/config.h"
 #include "storage/options.h"
+#include "storage/storage_engine.h"
+#include "storage/utils.h"
 
 namespace starrocks::spill {
 
 Status DirManager::init() {
-    std::vector<starrocks::StorePath> storage_paths;
-    RETURN_IF_ERROR(parse_conf_store_paths(config::spill_local_storage_dir, &storage_paths));
-    if (storage_paths.empty()) {
+    std::vector<starrocks::StorePath> spill_local_storage_paths;
+    RETURN_IF_ERROR(parse_conf_store_paths(config::spill_local_storage_dir, &spill_local_storage_paths));
+    if (spill_local_storage_paths.empty()) {
         return Status::InvalidArgument("cannot find spill_local_storage_dir");
     }
 
-    for (const auto& path : storage_paths) {
+    auto storage_paths = starrocks::StorageEngine::instance()->get_store_paths();
+    std::set<std::string> storage_path_set(storage_paths.begin(), storage_paths.end());
+
+    for (auto iter = spill_local_storage_paths.begin(); iter != spill_local_storage_paths.end();) {
+        const auto& path = iter->path;
+        if (storage_path_set.find(path) != storage_path_set.end()) {
+            return Status::InvalidArgument(fmt::format(
+                    "spill_local_storage_dir {} already exists in storage_root_path, please use another path", path));
+        }
+        if (!starrocks::check_datapath_rw(path)) {
+            if (starrocks::config::ignore_broken_disk) {
+                LOG(WARNING) << fmt::format("read write test spill_local_storage_dir {} failed, ignore it", path);
+                iter = spill_local_storage_paths.erase(iter);
+            } else {
+                return Status::IOError(
+                        fmt::format("read write test spill_local_storage_dir {} failed, please make sure it is "
+                                    "available and BE has permission to access",
+                                    path));
+            }
+        } else {
+            iter++;
+        }
+    }
+    if (spill_local_storage_paths.empty()) {
+        return Status::InvalidArgument("cannot find available spill_local_storage_dir");
+    }
+
+    std::regex query_id_pattern("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$");
+
+    for (const auto& path : spill_local_storage_paths) {
```

Comment:
This will make the BE start very slowly

Reply:
> This will make the BE start very slowly

i don't think so…
1. users needn't configure a lot of paths in  spill_local_storage_dir. One path for each disk is enough. I think the number of paths is at most a dozen
2. under normal circumstances, the directory generated by spilling will be deleted after the query is completed. There will be leftover directories only when a crash occurs.  These sub dirs need to be processed when BE starts, and I think its number won't be very large